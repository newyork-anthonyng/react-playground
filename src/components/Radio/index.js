import React, { Component } from "react";
import T from "prop-types";
import glamorous from "glamorous";

const CheckedRadio = glamorous.div(
  {
    "&::before": {
      border: "1px solid black",
      borderRadius: "100%",
      content: `''`,
      display: "inline-block",
      height: "10px",
      marginRight: "5px",
      width: "10px"
    }
  },
  props => ({
    "&::before": {
      backgroundColor: props.selected ? "tomato" : ""
    }
  })
);

class Radio extends Component {
  render() {
    const { selected, onClick, children } = this.props;

    return (
      <CheckedRadio
        role="radio"
        tabIndex={selected ? "0" : "-1"}
        aria-checked={selected ? "true" : "false"}
        selected={selected}
        onClick={onClick}
      >
        {children}
      </CheckedRadio>
    );
  }
}

Radio.defaultProps = {
  selected: false,
  onClick: () => {}
};

Radio.propTypes = {
  selected: T.bool.isRequired,
  children: T.node.isRequired,
  onClick: T.func.isRequired
};

export default Radio;
