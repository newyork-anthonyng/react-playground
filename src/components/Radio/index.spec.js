import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Radio from "./";

const defaultProps = {
  selected: false,
  onClick: () => {}
};

it("should render correctly when selected", () => {
  const wrapper = shallow(
    <Radio {...defaultProps} selected>
      <h1>Hello</h1>
    </Radio>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should render correctly when not selected", () => {
  const wrapper = shallow(
    <Radio {...defaultProps} selected={false}>
      <h1>Hello</h1>
    </Radio>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should run callback when clicked", () => {
  const cb = jest.fn();
  const wrapper = shallow(
    <Radio {...defaultProps} onClick={cb}>
      <h1>Hello</h1>
    </Radio>
  );

  wrapper.simulate("click");

  expect(cb).toHaveBeenCalledTimes(1);
});
