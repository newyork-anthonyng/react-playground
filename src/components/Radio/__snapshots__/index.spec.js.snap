// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when not selected 1`] = `
<glamorous(div)
  aria-checked="false"
  onClick={[Function]}
  role="radio"
  selected={false}
  tabIndex="-1"
>
  <h1>
    Hello
  </h1>
</glamorous(div)>
`;

exports[`should render correctly when selected 1`] = `
<glamorous(div)
  aria-checked="true"
  onClick={[Function]}
  role="radio"
  selected={true}
  tabIndex="0"
>
  <h1>
    Hello
  </h1>
</glamorous(div)>
`;
